ImperatorToCK3/Source/Imperator/Characters/PortraitData.cpp

```diff
@@ -0,0 +1,50 @@
+#include "PortraitData.h"
+#include "Log.h"
+#include "ParserHelpers.h"
+#include "base64.h"
+#include <bitset>
+
+
+long long binaryToDecimal(long long n)
+{
+	long long num = n;
+	long long dec_value = 0;
+
+	// Initializing base value to 1, i.e 2^0 
+	int base = 1;
+
+	long long temp = num;
+	while (temp) {
+		int last_digit = int(temp % 10);
+		temp = temp / 10;
+
+		dec_value += long long(last_digit * base);
+
+		base = base * 2;
+	}
+
+	return dec_value;
+}
+
+
+ImperatorWorld::CharacterPortraitData::CharacterPortraitData(std::string theString)
+{
+	const std::string& decodedDnaStr = base64_decode(theString);
+
+	std::string binary_outputInformations;
+	for (std::size_t i = 0; i < decodedDnaStr.size(); ++i)
```

Comment:
use auto, it's easier on the compiler and later reading.

Reply:
✔️

Reply:
I had to return to `std::size_t` because using `auto` caused a warning.